File: B8_RestFragmentsAsync/WeatherReportApp/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportapp/fragments/WeatherDetailsFragment.java
package com.example.tsvetelinastoyanova.weatherreportapp.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.tsvetelinastoyanova.weatherreportapp.Constants;
import com.example.tsvetelinastoyanova.weatherreportapp.ImageOperator;
import com.example.tsvetelinastoyanova.weatherreportapp.R;
import com.example.tsvetelinastoyanova.weatherreportapp.models.multiple.cities.model.WeatherObject;


public class WeatherDetailsFragment extends Fragment {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_weather_details, container, false);
    }

    public void changeView(WeatherObject object){
        TextView cityAndCountry = getView().findViewById(R.id.city_and_country);
        String city = object.getName();
        String country = object.getSys().getCountry();
        cityAndCountry.setText(String.format(Constants.COUNTRY_CITY,city, country));

        ImageView icon = getView().findViewById(R.id.icon);
        int id = ImageOperator.getImageIdFromString(object.getWeather().get(0).getIcon());
        icon.setImageDrawable(getActivity().getResources().getDrawable(id));

        TextView degrees =  getView().findViewById(R.id.degrees);
        degrees.setText(String.format(Constants.DEGREES_CELSIUS,object.getMain().getTemp()));

        TextView weather = getView().findViewById(R.id.weather);
        weather.setText(object.getWeather().get(0).getMain());

        TextView tempMin =  getView().findViewById(R.id.temp_min);
        tempMin.setText(getResources().getString(R.string.temperature_min,String.format(Constants.DEGREES_CELSIUS,object.getMain().getTempMin())));

        TextView tempMax =  getView().findViewById(R.id.temp_max);
        tempMax.setText(getResources().getString(R.string.temperature_max,String.format(Constants.DEGREES_CELSIUS,object.getMain().getTempMax())));

        TextView pressure =  getView().findViewById(R.id.pressure);
        pressure.setText(getResources().getString(R.string.pressure, object.getMain().getPressure()));

        TextView humidity =  getView().findViewById(R.id.humidity);
        humidity.setText(getResources().getString(R.string.humidity,object.getMain().getHumidity()));

        TextView windSpeed =  getView().findViewById(R.id.wind_speed);
        windSpeed.setText(getResources().getString(R.string.wind_speed,object.getWind().getSpeed()));

    }

}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/weatherdetails/weather/WeatherDetailsFragment.kt
package com.example.tsvetelinastoyanova.weatherapp.weatherdetails

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.tsvetelinastoyanova.weatherapp.Constants
import com.example.tsvetelinastoyanova.weatherapp.R
import com.example.tsvetelinastoyanova.weatherapp.model.currentweather.CurrentWeatherObject
import com.example.tsvetelinastoyanova.weatherapp.util.ImageOperator
import com.example.tsvetelinastoyanova.weatherapp.util.Utils
import com.example.tsvetelinastoyanova.weatherapp.weatherdetails.weather.WeatherDetailsContract
import kotlinx.android.synthetic.main.fragment_weather_details.*


class WeatherDetailsFragment : Fragment(), WeatherDetailsContract.View {

    private var presenter: WeatherDetailsContract.Presenter? = null

    /*** interface ***/

    private var onLoadedChildFragmentDelegate: OnLoadedChildFragmentDelegate? = null

    interface OnLoadedChildFragmentDelegate {
        fun onLoadedWeatherFragment()
    }

    fun setDelegate(delegate: OnLoadedChildFragmentDelegate) {
        onLoadedChildFragmentDelegate = delegate
    }

    /*** Methods from Fragment ***/

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_weather_details, container, false)
        return view
    }

    override fun setPresenter(presenter: WeatherDetailsContract.Presenter) {
        this.presenter = presenter
    }

    override fun onResume() {
        super.onResume()
        presenter?.let {
            it.start()
        }
        Utils.checkNotNull(onLoadedChildFragmentDelegate)
        onLoadedChildFragmentDelegate?.onLoadedWeatherFragment()
    }

    override fun changeView(weatherObject: CurrentWeatherObject) {
        val city = weatherObject.name
        val country = weatherObject.sys.country
        cityAndCountry.text = String.format(Constants.COUNTRY_CITY, city, country)

        val id = ImageOperator.getImageIdFromString(weatherObject.weather[0].icon)
        icon.setImageDrawable(ContextCompat.getDrawable(requireContext(), id))

        degrees.text = String.format(Constants.DEGREES_CELSIUS, weatherObject.main.temp)
        weather.text = weatherObject.weather[0].description
        tempMin.text = resources.getString(R.string.temperature_min, String.format(Constants.DEGREES_CELSIUS, weatherObject.main.tempMin))
        tempMax.text = resources.getString(R.string.temperature_max, String.format(Constants.DEGREES_CELSIUS, weatherObject.main.tempMax))
        pressure.text = resources.getString(R.string.pressure, weatherObject.main.pressure)
        humidity.text = resources.getString(R.string.humidity, weatherObject.main.humidity)
        windSpeed.text = resources.getString(R.string.wind_speed, weatherObject.wind.speed)
    }

    companion object {
        fun newInstance(): WeatherDetailsFragment {
            return WeatherDetailsFragment()
        }
    }
}
